incubator/hnc/internal/reconcilers/object.go

```diff
@@ -658,15 +657,21 @@ func (r *ObjectReconciler) syncPropagation(ctx context.Context, log logr.Logger,
 // - Objects have a selector that doesn't match the destination namespace
 // - Service Account token secrets
 func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstructured.Unstructured, dst string) bool {
-	selector, err := selectors.GetSelector(inst)
-	if err != nil {
-		log.Error(err, "Invalid value")
+	selector, errSelector := selectors.GetSelector(inst)
+	// TODO: generate events for the errors
+	if errSelector != nil {
+		log.Error(errSelector, "Cannot propagate")
 	}
-	treeSelector, err := selectors.GetTreeSelector(inst)
-	if err != nil {
-		log.Error(err, "Invalid value")
+	treeSelector, errTreeSelector := selectors.GetTreeSelector(inst)
+	if errTreeSelector != nil {
+		log.Error(errTreeSelector, "Cannot propagate")
+	}
+	noneSelector, errNoneSelector := selectors.GetNoneSelector(inst)
+	if errNoneSelector != nil {
+		log.Error(errNoneSelector, "Cannot propagate")
+		// We set noneSelector to true to protect any object in the child namespaces to be overwritten
+		noneSelector = true
```

Comment:
I don't think we need this anymore? Because if `errNoneSelector` is non-nil, we'll `return false` before we ever check the value of this.

Reply:
done